### PR TITLE
Contributor updates for repo-specific documentation

### PR DESCRIPTION
**Pull request by erogluorhan**
This PR updates `README.md` and `CONTRIBUTING.md` as well as adds an `INSTALLATION.md` to provide information about installation, build, and contribution that are specific to this repo. It also updates the environment files under `/build_envs` to differ conda environment `geocat_build` from that of `GeoCAT-comp` with the same name.

@anissa111 Please note that RTD files under `/docs` directory should be updated after this PR is merged in order to have RTD link to repo-specific files here.

ALL: Please have your suggestions as inline comments to the files changed, then I will incorporate them ASAP to the documents.

### CODE REVIEW

**Comment by erogluorhan** [comment]
> Looks good, Orhan. I think having Anissa work through a function implementation will shake out any inaccuracies or unclear information.

Yes, that makes much sense.

File: .circleci/build-libncomp.sh [hunk 1]
```diff
@@ -13,6 +14,17 @@ git checkout ${CIRCLE_BRANCH} || echo "No ${CIRCLE_BRANCH} on libncomp"
...
 make install
+
+# Test the build
+${NM:-nm} ${CONDA_PREFIX}/lib/libncomp.a
+if [ $(uname) = Darwin ]; then
+    ${OTOOL:-otool} -L ${CONDA_PREFIX}/lib/libncomp.dylib
+else
+    ${LDD:-ldd} ${CONDA_PREFIX}/lib/libncomp.so
+fi
+make check
```
Comment (pilotchute) [inline .circleci/build-libncomp.sh:30]: Where did this test boilerplate come from?
 It looks good.

File: docs/installation.rst [hunk 2]
```diff
@@ -54,13 +54,13 @@ installed.
...
 
-    - Python 3.5+
+    - Python 3.6+
     - numpy
     - xarray
     - cython
     - dask
     - pytest
-    - `ncomp <http://github.com/NCAR/ncomp/>`_
+    - `libncomp <https://github.com/NCAR/libncomp/>`_
```
Comment (pilotchute) [inline docs/installation.rst:63]: Looks good

File: .circleci/build-libncomp.sh (hunk 1 above)
Comment (erogluorhan) [reply .circleci/build-libncomp.sh:30]: It is now inherited from `libncomp` repo but I have no idea where it was initially originated from.

**Pull request merged**